js/models/SMTPIntegrationSettings.js

```diff
@@ -0,0 +1,43 @@
+import BaseModel from './BaseModel';
+import app from '../app';
+
+export default class extends BaseModel {
+  defaults() {
+    return {
+      smtpNotifications: false,
+      smtpServerAddress: 'defaultservername',
+      smtpUserName: '',
+      smtpPassword: '',
+      smtpFromEmail: '',
+      smtpToEmail: ''
+    };
+  }
+
+  validate(attrs) {
+    const errObj = {};
+    const addError = (fieldName, error) => {
+      errObj[fieldName] = errObj[fieldName] || [];
+      errObj[fieldName].push(error);
+    };
+
+    if (attrs.smtpNotifications) {
+      if ( attrs.smtpServerAddress.trim().length == 0 )
```

Comment:
For whatever code editor you are using, there is almost certainly an es-lint plugin which would highlight issues right there in the editor. Josh and I both use one and I've found it to be super helpful.